### PR TITLE
Add Stats for jvm cpu usage

### PR DESCRIPTION
**Pull request by nishantmonu51**
Add stats for JVM CPU Usage
- Useful when multiple JVMs are hosted on a single Machine.
- e.g For Druid MM nodes - these will give an idea of cpu usage per
peon process.

### CODE REVIEW

**Comment by drcrallen** [comment]
uuuuuuuniiiittttt tessssstttsssss :)

File: src/main/java/com/metamx/metrics/JvmMonitor.java [hunk 1]
```diff
@@ -112,6 +122,30 @@ public boolean doMonitor(ServiceEmitter emitter)
       emitter.emit(builder.build("jvm/bufferpool/count", pool.getCount()));
     }
 
+    // process CPU
+    try {
+      ProcCpu procCpu = sigar.getProcCpu(sigar.getPid());
```
Comment (drcrallen) [inline src/main/java/com/metamx/metrics/JvmMonitor.java]: Can you just use the JVM stats here, and not rely on sigar?

**Comment by xvrl** [comment]
@nishantmonu51 my only concern is that currently JvmMonitor does not require sigar, and this would make sigar a dependency for JvmMonitor as well, breaking anything that uses it.

**Comment by nishantmonu51** [comment]
@xvrl : we have the following ways for this -  
1) Use any java alternative - not sure if there is any. 
2) Add these stats to a new monitor e.g. JVMCpuMonitor which depends on sigar.  
3) Add these stats to existing SysMonitor which already depends on sigar 

I will prefer above steps in the order they are listed. 
What are your views on it ?

File: src/main/java/com/metamx/metrics/JvmMonitor.java (hunk 1 above)
Comment (nishantmonu51) [inline src/main/java/com/metamx/metrics/JvmMonitor.java]: @drcrallen can you point me to the JVM stats where i can get this info ?

**Comment by xvrl** [comment]
@nishantmonu51 if it's possible to move those metrics to SysMonitor, that would be nice.

**Comment by xvrl** [comment]
I'm also ok with a separate monitor.

**Comment by nishantmonu51** [comment]
closing until comments are handled.

**Pull request closed without merging**